lib/api/controllers/baseController.js

```diff
@@ -67,137 +69,427 @@ class BaseController {
-   * @param {String} paramPath - Path of the parameter to extract (eg: 'body.foo')
-   * @param {?Object} defaultValue
+   * @param {object} obj - container object
+   * @param {string} protocol - network protocol name
+   * @param {string} name - parameter name
+   * @param {string} errorName - name to use in error messages
+   * @return {boolean}
+   */
+  _getBoolean (obj, protocol, name, errorName) {
+    let value = obj[name];
+
+    // In HTTP, booleans are flags: if it's in the querystring, it's set,
+    // whatever its value.
+    // If a user needs to unset the option, they need to remove it from the
+    // querystring.
+    if (protocol === 'http') {
+      value = value !== undefined;
+      obj[name] = value;
+    }
+    else if (value === undefined || value === null) {
+      value = false;
+    }
+    else if (typeof value !== 'boolean') {
+      assertionError.throw('invalid_type', errorName, 'boolean');
+    }
+    else {
+      value = Boolean(value);
+    }
+
+    return value;
+  }
+
+  /**
+   * Generic object getter: number value
    *
-   * @throws
-   * @returns {Object}
+   * @param {object} obj - container object
+   * @param {string} name - parameter name
+   * @param {string} errorName - name to use in error messages
+   * @param {number} def - default value
+   * @return {number}
    */
-  getObjectParam (request, paramPath, defaultValue = null) {
-    const objectParam = _.get(request.input, paramPath) || defaultValue;
+  _getNumber (obj, name, errorName, def = null) {
+    let value = obj[name];
+
+    if (value === undefined || value === null) {
+      if (def !== null) {
+        return def;
+      }
 
-    if (objectParam === null) {
-      errorsManager.throw('missing_param', paramPath);
+      assertionError.throw('missing_argument', errorName);
     }
 
-    if (! _.isPlainObject(objectParam)) {
-      errorsManager.throw('invalid_param_type', paramPath, objectParam, 'Object');
+    value = Number.parseFloat(value);
+
+    if (Number.isNaN(value)) {
+      assertionError.throw('invalid_type', errorName, 'number');
     }
 
-    return objectParam;
+    return value;
   }
 
   /**
-   * Extracts a number parameter from the request input
+   * Generic object getter: integer value
    *
-   * @param {Request} request - Request object
-   * @param {String} paramPath - Path of the parameter to extract (eg: 'body.foo')
-   * @param {?Number} defaultValue
+   * @param {object} obj - container object
+   * @param {string} name - parameter name
+   * @param {string} errorName - name to use in error messages
+   * @param {integer} def - default value
+   * @return {integer}
+   */
+  _getInteger (obj, name, errorName, def = null) {
+    let value = obj[name];
+
+    if (value === undefined || value === null) {
+      if (def !== null) {
+        return def;
+      }
+
+      assertionError.throw('missing_argument', errorName);
+    }
+
+    value = Number.parseFloat(value);
```

Comment:
Shouldn't be `parseInt` ?

Reply:
`parseInt` truncates numbers to transform them into integers:

```
> Number.parseInt('3.14')
3
```

We don't want to change provided values silently, so we need to parse the number as a floating-point value, and then verify that it's an integer number.